pkg/manager/internal.go

```diff
@@ -188,6 +209,40 @@ func (cm *controllerManager) SetFields(i interface{}) error {
 	return nil
 }
 
+// AddHealthzCheck allows you to add Healthz checker
+func (cm *controllerManager) AddHealthzCheck(name string, check healthz.Checker) error {
+	cm.mu.Lock()
+	defer cm.mu.Unlock()
+
+	if cm.healthzStarted {
+		return fmt.Errorf("unable to add new checker because healthz endpoint has already been created")
+	}
+
+	if cm.healthzHandler == nil {
+		cm.healthzHandler = &healthz.Handler{Checks: map[string]healthz.Checker{}}
+	}
+
+	cm.healthzHandler.Checks[name] = check
```

Comment:
:woman_shrugging: we don't have a way to replace otherwise, so I'm not sure.  Let's leave it like this for the moment, and see what people do with it.